system/Database/SQLite3/Connection.php

```diff
@@ -107,7 +107,7 @@ public function connect(bool $persistent = false)
                 $this->database = WRITEPATH . $this->database;
             }
 
-            $sqlite = (! isset($this->password) || $this->password !== '')
+            $sqlite = ($this->password === null || $this->password !== '')
```

Comment:
@michalsn quick question. should the RHS of this conditional be checking `=== ''`?

Reply:
I think this should be handled in a separate PR, so I'll hold off on the fix until this one is merged.

Reply:
Thanks for confirming. I think this is good to be merged. Merging.